### PR TITLE
Added suggester to complete input multiply times by last entered word

### PR DESCRIPTION
**Pull request by lakdemon**
**Please review the following checklist.**

- [x] Docstrings on all new or modified functions / classes 
- [ ] Updated documentation
- [ ] Updated CHANGELOG.md (where appropriate)

### CODE REVIEW

**Comment by rodrigogiraoserrao** [comment]
Hey @lakdemon, thanks for your PR.
Do you mind giving me a bit more of context as to what your PR is trying to do, what the rationale for adding this is, etc?

Thanks!

**Comment by lakdemon** [comment]
I use this class in my application and I think it might be useful to have it in textual.
The matter is that SuggestFromList can add a phrase only once at the beginning of input.
For example:
**"S"**_elect something from ..._
**"I"**_nput data into ..._
This class can offer a hint for each word as you type.
For example dictionary name
**"S"**_elect_
**"Select f"**_rom_
**"Select from t"**_able_

This is much more convenient in some cases and may come in handy later for TextArea inputs i think.
If someone is interested in this, then I can try to add tests and a description for the PR

**Comment by lakdemon** [comment]
suggestion_example

**Comment by rodrigogiraoserrao** [comment]
@willmcgugan I need your input on this!
Pun intended!

**Comment by rodrigogiraoserrao** [comment]
> I'm afraid I don't quite grasp what this is doing over the existing suggester. I would need example(s)

See the GIF here: https://github.com/Textualize/textual/pull/2881#issuecomment-1622640222

This suggester only completes the last word whereas the suggester that is already implemented completes the whole input.

**Comment by willmcgugan** [comment]
Sounds like a good idea in principle. The PR would need a bit of work to go in to the core library.

File: src/textual/suggester.py [hunk 1]
```diff
@@ -136,3 +136,19 @@ async def get_suggestion(self, value: str) -> str | None:
...
+class SuggestLastWordFromList(SuggestFromList):
+    """
+        Give suggestions for completing each last word in the text based on a fixed list of options.
+    """
+
+    async def get_suggestion(self, value: str) -> str | None:
+        if not value.endswith(' '):
+            values = value.rsplit(' ', 1)
+            last_word = values[-1]
+            for idx, suggestion in enumerate(self._for_comparison):
```
Comment (willmcgugan) [inline src/textual/suggester.py:150]: Probable shouldn't be using "private" data here. Suggest directly suggesting from `Suggester`.

**Comment by rodrigogiraoserrao** [comment]
> The PR would need a bit of work to go in to the core library.

I'm happy to work with @lakdemon to get the PR in a good state for us to review.

File: src/textual/suggester.py (hunk 1 above)
Comment (willmcgugan) [inline src/textual/suggester.py:150]: We have a policy of no abbreviations, so `index` would be preferred over `id`

File: src/textual/suggester.py [hunk 2]
```diff
@@ -136,3 +136,19 @@ async def get_suggestion(self, value: str) -> str | None:
             if suggestion.startswith(value):
                 return self._suggestions[idx]
         return None
+
+
+class SuggestLastWordFromList(SuggestFromList):
+    """
```
Comment (willmcgugan) [inline src/textual/suggester.py:142]: Docstring could use more detail. See other docstrings for inspiration.

**Review by willmcgugan — CHANGES_REQUESTED** [review]

**Comment by rodrigogiraoserrao** [comment]
Hey @lakdemon, are you happy to go over the requested changes or do you want to close this PR?

**Comment by willmcgugan** [comment]
Closing for now. Assumed stale.

My gut feeling is that it doesn't necessarily belong in the core library.

**Pull request closed without merging**